### PR TITLE
chore(tests): add podman desktop update e2e test

### LINKED ISSUE #7931: Implement the E2E update path of the Podman Desktop
### Is your enhancement related to a problem? Please describe

We are seeing a problem with electron when updating the podman desktop to the latest released version on windows: https://github.com/containers/podman-desktop/issues/7114.

### Describe the solution you'd like

We need to came up with a automated test that will make sure we know if we can actually perform the update or not.

* checkout main upstream branch
* change the version in `package.json` to 1.0.0
* build the app from the codebase
* start up the e2e tests (testing the update path)
* check that we have option to update to the newest podman-desktop version (latest tag on github) - which is actually a version below what we are testing
* update
* check that update was sucessfull
* We might end the test once we get "Do you want to restart?" dialog, since we could probably not be able to let app to get restarted (test would crash)
* Alternative would be to start another instance in the next step and check the version and let GHA/CI to evaluate the result, not the tests

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Brings new E2E update test. The test is intended to be run solely in the specific test environment. Windows (now) and Mac will be target env. 

The update e2e test will run after each merge, and in the PR check if it was initialized by dependabot where we saw the most problems.

### What issues does this PR fix or reference?
#7931 

### How to test this PR?
On windows only now. See the details in the linked issue. You need change version of PD in package.json to something like `1.0.0` and then run `yarn test:e2e:update`. Also, if you want to perform actual update during the testing, add env. var: `UPDATE_PODMAN_DESKTOP=true`
In order to be able to run the update in locally built podman desktop, you need to set `ELECTRON_ENABLE_INSPECT` env. var. and build the bits to have correct artifacts: `pnpm compile:current --win nsis`. It will produce executable podman desktop binary, which will offer an update option. In order to execute the e2e tests now with built executable, use `PODMAN_DESKTOP_BINARY="C:\path\to\dist\win-unpacked\Podman Desktop.exe"` when running test target.

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by odockal** [comment]
@ScrewTSW In this pr there is the example of how to build and use the compiled binary with playwright.

**Review by xbabalov — APPROVED** [review]

File: tests/playwright/src/specs/image-smoke.spec.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ test.beforeAll(async () => {
 });
 
 test.afterAll(async () => {
+  // test
```
Comment (cbr7) [inline tests/playwright/src/specs/image-smoke.spec.ts]: Does this comment need to be here? Maybe it's an artifact from some previous debugging?

**Review by cbr7 — APPROVED** [review]

File: .github/workflows/pr-check-update-e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  pull_request:
+    paths:
+      - '**'
+      - 'pnpm-lock.yaml'
+    types: [labeled, synchronize, opened, reopened]
+
+jobs:
+  detect_pnpm_changes:
+    name: Detect pnpm lock file changes
+    runs-on: ubuntu-latest
```
Comment (benoitf) [inline .github/workflows/pr-check-update-e2e.yaml]:
```suggestion
    runs-on: ubuntu-24.04
```

File: .github/workflows/pr-check-update-e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            echo "PNPM_LOCK_CHANGED=true" >> $GITHUB_OUTPUT
+          else
+            echo "PNPM_LOCK_CHANGED=false" >> $GITHUB_OUTPUT
+          fi
+
+  run-update-e2e-test:
+    name: win update e2e tests
+    needs: detect_pnpm_changes
+    if: contains(github.event.pull_request.labels.*.name, 'area/update') || needs.detect_pnpm_changes.outputs.pnpm_lock_changed == 'true'
+    runs-on: ${{ matrix.os }}
```
Comment (benoitf) [inline .github/workflows/pr-check-update-e2e.yaml]: why do we have a matrix if title and os are hardcoded to windows ?

Comment (benoitf) [reply .github/workflows/pr-check-update-e2e.yaml]: if matrix is used, name should use matrix.os and not win

Comment (odockal) [reply .github/workflows/pr-check-update-e2e.yaml]: I was a step ahead, will keep it for windows for now.

File: .github/workflows/pr-check.yaml [hunk 4]
```diff
@@ -324,3 +326,98 @@ jobs:
...
+    steps:
+      - name: Evaluate the Windows Update test results
+        run: | 
+          echo "Windows updater result: ${{ needs.run-update-e2e-test.result }}"
+          if [ "${{ needs.run-update-e2e-test.result }}" = "failure" ]; then
+            echo "Windows udpater test failed..."
+            exit 1
+          else
+            echo "Windows updater test succeeded or was skipped..."
+          fi
```
Comment (ScrewTSW) [inline .github/workflows/pr-check.yaml]: missing new line

**Review by ScrewTSW — APPROVED** [review]
LGTM

**Review by benoitf — APPROVED** [review]

File: .github/workflows/pr-check.yaml [hunk 5]
```diff
@@ -324,3 +326,98 @@ jobs:
...
+        with:
+          name: win-update-e2e-test
+          path: |
+            ./tests/**/output/
+            ./tests/**/output/junit*.xml
+            !./tests/**/traces/raw
+
+  update-e2e-test:
+    name: update E2E test
+    runs-on: ubuntu-22.04
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: why not 24 04

File: tests/playwright/src/specs/installation/update-install.spec.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
+/**********************************************************************
+ * Copyright (C) 2023-2024 Red Hat, Inc.
```
Comment (benoitf) [inline tests/playwright/src/specs/installation/update-install.spec.ts]:
```suggestion
 * Copyright (C) 2024 Red Hat, Inc.
```

File: .github/workflows/pr-check.yaml (hunk 5 above)
Comment (odockal) [reply .github/workflows/pr-check.yaml]: mistake, updated.

File: tests/playwright/src/specs/installation/update-install.spec.ts (hunk 6 above)
Comment (odockal) [reply tests/playwright/src/specs/installation/update-install.spec.ts]: fixed, thx.

**Pull request merged**